File: handler.js
const child_process = require('child_process');
const AWS = require('aws-sdk');
const docClient = new AWS.DynamoDB.DocumentClient();


exports.convert = async (event) => {
  let requestid = event.requestContext.requestId;
  let ip = event.requestContext.identity.sourceIp;
  let documentUrl = event.queryStringParameters.document_url;

  await docClient.put({
      TableName: process.env.TABLE_NAME,
      Item: {
        'requestid': requestid,
        'ip': ip,
        'document_url': documentUrl
      }
    }
  ).promise();
  
  let txt = child_process.execSync(`curl --silent -L ${documentUrl} | ./bin/catdoc -`).toString();

  return {
    statusCode: 200,
    body: txt
  };
};

File: handler.py
import subprocess
import boto3
import os

dynamodb = boto3.client("dynamodb")


def convert(event, context):
    requestid = event["requestContext"]["requestId"]
    ip = event["requestContext"]["identity"]["sourceIp"]
    document_url = event["queryStringParameters"]["document_url"]

    dynamodb.put_item(
        TableName=os.environ["TABLE_NAME"],
        Item={
            'requestid': {"S": requestid},
            'ip': {"S": ip},
            'document_url': {"S": document_url}
        }
    )

    txt = subprocess.check_output("curl --silent -L %s | ./bin/catdoc -" % document_url, shell=True)

    return {
        "statusCode": 200,
        "body": txt
    }
